plugin/src/main/java/org/opensearch/ml/rest/RestMLUpdateConnectorAction.java

```diff
@@ -68,9 +69,12 @@ private MLUpdateConnectorRequest getRequest(RestRequest request) throws IOExcept
 
         String connectorId = getParameterId(request, PARAMETER_CONNECTOR_ID);
 
-        XContentParser parser = request.contentParser();
-        ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.nextToken(), parser);
-
-        return MLUpdateConnectorRequest.parse(parser, connectorId);
+        try {
+            XContentParser parser = request.contentParser();
+            ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.nextToken(), parser);
+            return MLUpdateConnectorRequest.parse(parser, connectorId);
+        } catch (IllegalStateException illegalStateException) {
+            throw new OpenSearchParseException(illegalStateException.getMessage());
```

Comment:
I can confirm in line 67 `IOException`  could also throw a 500 status code, would you mind changing the `throw new IOException` to `throw new OpenSearchParseException`?

Reply:
+1 on previous comment. Don't we need to add corresponding test as well?

Reply:
This is a good catch. Already updated in the new commit. The UT tests are added too.